search-commons/src/main/java/no/unit/nva/search2/common/AggregationFormat.java

```diff
@@ -43,9 +45,36 @@ public static JsonNode apply(JsonNode aggregations) {
         getAggregationFieldStreams(aggregations)
             .map(AggregationFormat::getJsonNodeEntry)
             .forEach(item -> objectNode.set(item.getKey(), fixNodes(item.getValue())));
+        ensureChildNodesAreArrays(objectNode);
         return objectNode;
     }
 
+    public static void ensureChildNodesAreArrays(JsonNode node) {
+        if (node.isObject()) {
+            var objectNode = (ObjectNode) node;
+            objectNode.fieldNames().forEachRemaining(field -> processChildNode(objectNode, field));
+        }
+    }
+
+    private static void processChildNode(ObjectNode objectNode, String field) {
+        var childNode = objectNode.get(field);
+        if (childNode.isObject()) {
+            var nodeArray = JsonUtils.dtoObjectMapper.createArrayNode();
+            childNode.fieldNames().forEachRemaining(node -> processChildNodes(node, childNode, nodeArray));
+        objectNode.set(field, nodeArray);
+        }
+    }
+
+    private static void processChildNodes(String field, JsonNode node, ArrayNode nodeArray) {
+        if (node.get(field).isObject()) {
+            var childNode = node.get(field);
+            var newNode = JsonUtils.dtoObjectMapper.createObjectNode();
+            newNode.put(Constants.DOC_COUNT, childNode.get(Constants.DOC_COUNT).asInt());
+            newNode.put(KEY, field);
+            nodeArray.add(newNode);
+        }
+    }
```

Comment:
This is ugly, but it is the only way i found to serve "filter" as facet.

Reply:
Since it is only fixing one node, we could just select it and fix it, and update functionNames to reflect this.